### PR TITLE
c++ api: Some cleanup of documentation and declarations.

### PR DESCRIPTION
**Pull request by aniemetz**

### CODE REVIEW

File: include/cvc5/cvc5_kind.h [hunk 1]
```diff
@@ -10,23 +10,35 @@
...
 
 /* -------------------------------------------------------------------------- */
 /* Kind                                                                       */
 /* -------------------------------------------------------------------------- */
 
+#ifdef CVC5_API_USE_C_ENUMS
+#undef EVALUE
+#define EVALUE(name) CVC5_KIND_##name
+#endif
+
```
Comment (barrettcw) [inline include/cvc5/cvc5_kind.h:41]: Why is this here and not included in the first #ifdef CVC5_API_USE_C_ENUMS above?  Seems inconsistent with how EVALUE is defined in the #else case.

Comment (aniemetz) [reply include/cvc5/cvc5_kind.h:41]: It has to be redefined for each enum.

Comment (aniemetz) [reply include/cvc5/cvc5_kind.h]: Note that cvc5_kind.h is not touched in this PR.

**Review by mpreiner — APPROVED** [review]
